chore: added the used code from mendersoftware/mender

danielskinstad opened the pull request:
Mender-snapshot used github.com/mendersoftware/mender as a go module. The repo is not going to be updated further, and little code is used. Added the used code to this code base.

Changelog: title

Ticket: MEN-6896

lluiscampos commented on system/ioctl.go:
```diff
@@ -27,22 +26,17 @@ import (
-// See the linux kernel documentation: linux/Documentation/filesystems/proc.txt
-// A line takes the form:
-// 36 35 98:0 /mnt1 /mnt2 rw,noatime master:1 - ext3 /dev/root rw,errors=continue
-// (1)(2)(3)   (4)   (5)      (6)      (7)   (8) (9)   (10)         (11)
 type MountInfo struct {
```
This comment seems useful. I would restore it

lluiscampos commented on system/ioctl.go:
```diff
@@ -240,65 +195,39 @@ func getUbiDeviceSize(file *os.File) (uint64, error) {
-	return nil
+	return [2]uint32{^uint32(0), ^uint32(0)},
+		fmt.Errorf("invalid stat(2) st_mode %04X", devType)
 }
```
This file is really hard to review. As we have moved the functions around the diff shows many functions are removed and added. I looked and some of the functions to see that there were no code differences, so I assume that none of them had code differences :)

lluiscampos commented on go.sum:
```diff
@@ -1,1022 +1,43 @@
-honnef.co/go/tools v0.0.1-2020.1.4/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9vFzvIQ3k=
-rsc.io/binaryregexp v0.2.0/go.mod h1:qTv7/COck+e2FymRvadv62gMdZztPaShugOCi3I+8D8=
-rsc.io/quote/v3 v3.1.0/go.mod h1:yEA65RcK8LyAZtP9Kv3t0HmxON59tX3rD+tICJqUlj0=
-rsc.io/sampler v1.3.0/go.mod h1:T1hPZKmBbMNahiBKFy5HrXp6adAjACjK9JXDnKaTXpA=
```
I re-generated this locally on my machine and it looks good :+1:

lluiscampos reviewed (changes requested):
The code looks good, nice clean-up!

This commit is not a feature, so we should not mark it as such. Use `chore` better. Read [here](https://github.com/mendersoftware/mendertesting/blob/master/commitlint/grammar.md) to get more familiar with the categories that we use for our commits.

There are several failures in CI that you need to amend, like licenses or static checks. Let me know if you want assistance to fix them!
:point_right:  From the CI checks, one that was strange to me is that CI did not validate the signature of your commit because it has an extra space. Did you use the `-s` flag of `git commit` or did you type that yourself?

danielskinstad committed:
chore: added the used code from mendersoftware/mender

Mender-snapshot used github.com/mendersoftware/mender as a go module.
The repo is not going to be updated furhter, and little code is used.
Added the used code to this code base.

Ticket: MEN-6896

Signed-off-by: Daniel Skinstad Drabitzius <daniel.drabitzius@northern.tech>

lluiscampos reviewed (approved):
> This commit is not a feature, so we should not mark it as such. Use `chore` better. Read [here](https://github.com/mendersoftware/mendertesting/blob/master/commitlint/grammar.md) to get more familiar with the categories that we use for our commits.

I forgot to add that we should not have a changelog entry for this change neither. It does not add functionality nor fixes any bug. You can just remove the `changelog` line in your commit.

Otherwise looks good, pre-aproving.

Apart from fixing the commit, we need to run the integration pipeline :robot:

danielskinstad commented:
@mender-test-bot start pipeline
